hyperspy/drawing/widget.py

```diff
@@ -1020,3 +1020,52 @@ def _add_patch_to(self, ax):
             self._set_resizers(True, ax)
         if hasattr(super(ResizersMixin, self), '_add_patch_to'):
             super(ResizersMixin, self)._add_patch_to(ax)
+
+
+class MPLWidgetBase(WidgetBase):
```

Comment:
This class should also be a parent class of `RangeWidget`, which uses `matplotlib.widgets.SpanSelector`.

Would it make more sense that it inherit from `ResizableDraggableWidgetBase` instead of `WidgetBase`?

Reply:
There is a possibility for `MPLWidgetBase` to inherit from `ResizableDraggableWidgetBase `. However, then `PolygonWidget` will need to implement resizing and dragging, which can be possible, but not obvious. For example, how can the `PolygonWidget` be resized? So I can do it if it seems worth it

And I looked a bit into `RangeWidget` inheriting from it, but it seems to implement a lot from `ResizableDraggableWidgetBase`. So the question is how to do that without losing a bit of functionality. Maybe multiple inheritance?